source/common/http/http1/codec_impl.cc

```diff
@@ -379,23 +380,29 @@ http_parser_settings ConnectionImpl::settings_{
     },
     nullptr, // on_status
     [](http_parser* parser, const char* at, size_t length) -> int {
-      static_cast<ConnectionImpl*>(parser->data)->onHeaderField(at, length);
-      return 0;
+      Envoy::StatusOr<int> statusor =
+          static_cast<ConnectionImpl*>(parser->data)->onHeaderField(at, length);
+      return statusor.ok() ? statusor.value() : 1;
```

Comment:
Can you add a comment on why we return `1` in the non-OK case? Same elsewhere. You might consider a human readable constant for the special return codes if that is helpful for readability.

Reply:
Done. I'd prefer a human readable constant. For uniformity, I think it makes the most sense to use -1, and reserve 1 and 2 (which have special meanings for on_headers_complete)